spec/components/trainees/filters/view_spec.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Trainees::Filters::View do
+  let(:permitted_filters) { filters.permit(:subject, record_type: []) }
+  let(:result) { render_inline described_class.new(permitted_filters) }
+
+  context "when no filters are applied" do
+    let(:filters) { ActionController::Parameters.new({}) }
+
+    it "all of the checkboxes are unchecked" do
+      expect(result.css("#record_type-assessment_only").attr("checked")).to eq(nil)
+      expect(result.css("#record_type-provider_led").attr("checked")).to eq(nil)
+    end
+
+    it "does not show a 'Selected filters' dialogue" do
+      expect(result.text).not_to include("Selected filters")
+    end
+  end
+
+  context "when checkboxes have been pre-selected" do
+    let(:filters) do
+      ActionController::Parameters.new({ record_type: %w[assessment_only] })
+    end
+
+    it "marks the correct ones as selected" do
+      expect(result.css("#record_type-assessment_only").attr("checked").value).to eq("checked")
+      expect(result.css("#record_type-provider_led").attr("checked")).to eq(nil)
+    end
+
+    it "shows a 'Selected filters' dialogue" do
+      expect(result.text).to include("Selected filters")
```

Comment:
Might be worth putting "Selected filters" in a let and re-use across some of the examples. It'll remove the SonarCloud complaints